File: online-judge-solutions/Codeforces/1701A.py
#https://codeforces.com/problemset/problem/1701/A
#Lang		:	Python 3.8
#Time		:	30 ms
#Memory         :	0 KB
for _ in range(int(input())):
    n,s=4,0
    for _ in range(2):s+=(list(map(int,input().split()))[:2]).count(1)
    print(0 if s==0 else (2 if s==4 else 1))

File: online-judge-solutions/Codeforces/1701A.cpp
//https://codeforces.com/problemset/problem/1701/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	int t,x;
	cin>>t;
	while(t--){
		int n=4,s=0;
		while(n--){
			cin>>x;
			if(x==1) s++;
		}
		cout<<(s==0?"0":(s==4?"2":"1"))<<"\n";
	}
	return 0;
}
